File: backend/src/main/java/com/erp/backend_erp/dto/cattleSales/CreateCattleSaleDto.java
package com.erp.backend_erp.dto.cattleSales;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CreateCattleSaleDto {
    private String fechaVenta;
    private Double pesoTotal;
    private Double precioKilo;
    private String destino;
    private Long farmId;
    private List<Long> cattleIds;
    private Double precioTotal; 
    private String comprador;
    private List<CreateCattleSaleItemDto> items; 
}

File: frontend/src/app/modules/cattle-sales/domain/dto/create-cattle-sale.dto.ts
import { CreateCattleSaleItemDto } from './create-cattle-sale-item.dto';

export interface CreateCattleSaleDto {
  tipoVenta: 'INDIVIDUAL' | 'LOTE'; // Tipo de venta, puede ser 'INDIVIDUAL' o 'LOTE'
  numeroFactura: string;
  fechaVenta: string; // formato yyyy-MM-dd
  horaEmision: string; // formato HH:mm
  moneda: string;      // Ej: 'COP'
  formaPago: string;   // Ej: 'Contado', 'Crédito'
  
  destino: string;
  observaciones: string;

  precioKilo: number;
  pesoTotal: number;
  subtotal: number;
  iva: number;
  descuentos: number;
  total: number; // Total a pagar, puede ser igual a precioTotal

  terceroId: number;
  cattleIds: number[];
  items: CreateCattleSaleItemDto[];
}
